homeassistant/components/sonarr/calendar.py

```diff
@@ -70,16 +71,14 @@ async def async_get_events(
     ) -> list[CalendarEvent]:
         """Return calendar events within a datetime range."""
         episode_events = await get_sonarr_episode_events(self.coordinator)
-        if start_date and end_date:
-            start_date = start_date.replace(tzinfo=timezone.utc)
-            end_date = end_date.replace(tzinfo=timezone.utc)
-            self._events = [
-                episode
-                for episode in episode_events
-                if start_date <= episode.start <= end_date
-            ]
-        else:
-            self._events = episode_events
+        start_date = as_utc(start_date)
+        end_date = as_utc(end_date)
+        self._events = [
```

Comment:
We shouldn't modify the stored events here. The scheduled update should always the consider all the data.